### ISSUE TITLE
Error for instantiation of `Protocol`

### ISSUE THREAD

**Issue by tamuhey** [microsoft/pyright#1936]
I want pyright to report error for instantiation of `typing.Protocol`

```python
from typing import Protocol
class A(Protocol):
    def foo(self):
        ...

A()  # should raise error (mypy does)
```

thanks,

**Comment by erictraut** [comment]
When I run mypy on this sample, it doesn't give me an error. Is there a non-default option that I need to use?

**Comment by tamuhey** [comment]
I'm sorry the previous code is not appropriate. I've updated the code and now mypy reports the following error:

```
a.py:9: error: Cannot instantiate protocol class "A"
Found 1 error in 1 file (checked 1 source file)
```

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.147, which I just published. It will also be included in the next release of pylance.
